demisto_sdk/commands/validate/validators/BA_validators/BA109_file_name_has_separators.py

```diff
@@ -0,0 +1,107 @@
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Union[Integration, Script]
+ENTITY_NAME_SEPARATORS = ["_", "-"]
+
+
+class FileNameHasSeparatorsValidator(BaseValidator[ContentTypes]):
+    error_code = "BA109"
+    description = (
+        "Check if there are separators in the script or integration files names."
+    )
+    rationale = "Filenames for scripts and integrations should not contain separators to maintain consistency and readability."
+    error_message = "The {0} files {1} should be named {2}, respectively, without any separators in the base name."
+    related_field = "file path"
+    is_auto_fixable = False
+    invalid_files: List[str] = []
+    valid_files: List[str] = []
+
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message.format(
+                    content_item.content_type,
+                    " and ".join([f"'{word}'" for word in self.invalid_files]),
+                    " and ".join([f"'{word}'" for word in self.valid_files]),
+                ),
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if (self.check_separators_in_files(content_item))
+        ]
+
+    def check_separators_in_files(self, content_item: ContentTypes) -> bool:
+        """
+        Check if there are separators in the file names of the content item.
+
+        Args:
+            content_item: The content item to check.
+
+        Returns:
+            bool: True if there are invalid file names, False otherwise.
+        """
+        invalid_files: List[str] = []
+        valid_files: List[str] = []
+
+        files_to_check = get_files_in_dir(
+            str(content_item.path.parent), ["yml", "py", "md", "png"], False
+        )
+
+        files_to_check = sorted(
+            files_to_check
+        )  # Used to keep the error message consistent
+
+        for file_path in files_to_check:
+            if (file_name := Path(file_path).name).startswith("README"):
+                continue
+
+            if (
+                file_name.endswith("_image.png")
+                or file_name.endswith("_description.md")
+                or file_name.endswith("_test.py")
+                or file_name.endswith("_unified.yml")
+            ):
+                base_name = file_name.rsplit("_", 1)[0]
+            else:
+                base_name = file_name.rsplit(".", 1)[0]
+
+            valid_base_name = self.remove_separators_from_name(base_name)
+
+            if valid_base_name != base_name:
+                invalid_files.append(file_name)
+                valid_files.append(valid_base_name.join(file_name.rsplit(base_name, 1)))
+
+        if invalid_files:
+            self.invalid_files = invalid_files
+            self.valid_files = valid_files
+            return True
+
+        return False
```

Comment:
Let's save all the results in one list containing tuples of (valid_file, invalid file)
Also, return it from **check_separators_in_files** instead of a boolean flag and remove the new fields added to the class.

Reply:
I hope that's what you meant